src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -165,8 +165,12 @@ private void runEventLoop(SelectedSelectionKeySet keys)
       int size = tasks.size();
       try {
         if (size > 0) {
+          Runnable task;
           while (alive && size > 0) {
-            Runnable task = tasks.pollUnsafe();
+            task = tasks.poll();
+            if (task == null) {
```

Comment:
You may be correct especially since you have a test case. I am still wondering why that would happen. It's the event loop thread which is exclusively dealing with removing tasks from the queue. I'll take a look at the test case.